### DISCUSSION TITLE
Publish only selected files

### DISCUSSION THREAD

**Discussion by gaborschnierer** [Nice3point/RevitTemplates#47] [Q&A] [answered]
Hi Roman!

When I build my addin project, I get several dlls copied to the bin folder like 
System.Runtime.CompilerServices.Unsafe.dll 
System.Numerics.Vectors.dll
System.Buffers.dll
etc. totaling 11 items which look like they are not real dependencies, as the deployed addin works well without these files. 

To avoid distributing bloat, how would you recommend skipping these from my installer? I could maybe hard code exception file names or add a target in my Nuke project to delete these files at the right time, but maybe there are more elegant solutions like an MS Build config for that. I hoped you could quickly point me in a good direction with my search.

**Comment by Nice3point** [comment, marked as answer]
You are not using them directly, they are implicit dependencies of some packages you use.
Your plugin only works because another plugin probably loaded them earlier. If your plugin is the only one plugin in Revit, it probably won't run if you exclude these dependencies. 

In Revit 2025 and .Net 8, these libraries are part of the NET Runtime and so they won't be in the output.

My recommendation is to not remove them and keep them as is. To ensure that the plugin ships with all dependencies in use

Check your Nuget packages for these dependencies
![изображение](https://github.com/Nice3point/RevitTemplates/assets/20504884/2ad124b9-5945-4e3d-857c-2eaf6f1192fa)

**Reply by Nice3point** [reply]
But still if you want to exclude them, you can write your own target to publish the plugin. My old instruction (now deleted and not up to date) https://github.com/Nice3point/RevitTemplates/wiki/Third-party-files/e870074e6f6e0f8212d369e72bca793744712534

- Set `<DisableAddinBundleGeneration>true</DisableAddinBundleGeneration>` to disable current publish task
- Copy original publish task to your `.csproj` and edit it https://github.com/Nice3point/Revit.Build.Tasks/blob/main/source/Nice3point.Revit.Build.Tasks/targets/Nice3point.Revit.Publish.targets

**Reply by gaborschnierer** [reply]
Thanks, sure, I might just end up including them all just to make sure the shipped addin works. 
Still I really want to understand how these things work, and your answers are indeed pointing me in the right direction, as always 👍 Exploring the nuget.org dependencies and frameworks page was an eye-opening experience.

Here is what I still don't understand. Both the base Nice3point/RevitTemplates and my solution uses [CommunityToolkit.Mvvm](https://www.nuget.org/packages/CommunityToolkit.Mvvm/8.2.2). When I build mine the dependent dlls get copied in the bin folder. When I build yours, they don't get copied. This is the only package that lists these dlls as dependencies, and only under .NETStandard 2.0. Maybe my project uses the .NETStandard 2.0 for this dependency and yours uses .net6? How can this be controlled?

[Microsoft.Bcl.AsyncInterfaces](https://www.nuget.org/packages/Microsoft.Bcl.AsyncInterfaces/) (>= 7.0.0)
[System.ComponentModel.Annotations](https://www.nuget.org/packages/System.ComponentModel.Annotations/) (>= 5.0.0)
[System.Memory](https://www.nuget.org/packages/System.Memory/) (>= 4.5.5)
[System.Runtime.CompilerServices.Unsafe](https://www.nuget.org/packages/System.Runtime.CompilerServices.Unsafe/) (>= 6.0.0)

**Reply by Nice3point** [reply]
@gaborschnierer check your current configuration. 
![изображение](https://github.com/Nice3point/RevitTemplates/assets/20504884/c19c6e1c-df62-447d-8efa-b8335fb8f189)
As I wrote above, you won't find these libraries for Revit 2025, as Revit 2025 uses .Net 8 and they are included by default in runtime.

**2024:**
![изображение](https://github.com/Nice3point/RevitTemplates/assets/20504884/25210752-8015-41ef-aa5b-2dcf21b664d0)

**2025:**
![изображение](https://github.com/Nice3point/RevitTemplates/assets/20504884/e7a06113-09d7-46bd-9322-abaf0ecef080)

Maybe you're building a project for version 25 and that's why they're not there?
![изображение](https://github.com/Nice3point/RevitTemplates/assets/20504884/e7e30bb9-f9dc-4b93-aa29-a2bb22a282d9)
